Sources/TuistCore/Graph/Graph.swift

```diff
@@ -240,7 +240,7 @@ public class Graph: Encodable, Equatable {
 
         // Precompiled libraries and frameworks
 
-        let precompiledLibrariesAndFrameworks = targetNode.precompiledDependencies
+        let precompiledLibrariesAndFrameworks = targetNode.recursivePrecompiledDependencies
```

Comment:
I was wondering if we shall filter only dynamic transitive dependencies and the direct dependencies of a target. 

Thoughts? 🤔

Reply:
I don't think this is right. Your change is saying that we should link both, direct and transitive dependencies. That's not what users would expect.

Reply:
If we didn't link the transitive dependencies of `.framework`s in case of precompiled dependencies, we would get an error about missing frameworks. It is shown on the error screenshot in the issue description. Frameworks cannot embed other frameworks, so just adding already pre-built frameworks in `.tuist/Cache` to Framework Search Paths seemed to be a good idea 🤔